Code/Framework/AzCore/AzCore/o3dekernel_files.cmake

```diff
@@ -0,0 +1,18 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+set(FILES
+    Module/Environment.cpp
+    Module/Environment.h
+)
+
+if(NOT LY_MONOLITHIC_GAME)
+    list(APPEND FILES 
+        std/hash.cpp
+    )
+endif()
```

Comment:
We could just take the `hash_next_bucket_size` function in hash.cpp and move it to hash.h and avoid the need to selectively include a cpp file for the O3DEKernel